.github/workflows/build-prep.yml

```diff
@@ -18,6 +18,10 @@ jobs:
           cache: 'npm'
       - name: Install npm dependencies
         run: npm ci
+      - name: Run formatter
+        run: npm run format
```

Comment:
I don't see a reason to format the code just for machine consumption (just before builds) - that also runs the risk of making it harder to debug, if the code gets transformed such that line numbers on the release don't match line numbers that are in Git.

If you want to enforce that the code was formatted before a PR gets accepted, then maybe we can add it just to the CI part, but not to release builds?

Reply:
> If you want to enforce that the code was formatted before a PR gets accepted, then maybe we can add it just to the CI part, but not to release builds?

Ha! Great minds think alike :) I agree with you. That was my first thought as well. However, I ran into a myriad of problems with it, so I went with the solution you commented on instead. See my lengthy comment above for a full explanation.

Reply:
Got it. Ok, let's try it like this.

One more thought: if you plugged the formatter into lint, do you still need two separate steps here?

Reply:
> One more thought: if you plugged the formatter into lint, do you still need two separate steps here?

Unfortunately, I only achieved partial plugginess :( For stuff that is checked by ESLint (`*.ts`, `*.html`), there is indeed no need to run a separate formatter step, because the errors are translated to ESLint.

For `*.js` (this can probably be solved) and `*.css` / `*.scss` there is no integration. The "proper" way to do it for CSS seems to be to integrate another linter that works with Prettier, that would be StyleLint. But I don't have the willpower to get into that. So as a middle ground, my idea was to at least have formatting for SCSS, which you get "for free" with Prettier. But you have to call it separately to enforce it.